packet/src/bwe/rate_controller.rs

```diff
@@ -0,0 +1,376 @@
+// TODO: Remove this when we integrated RateController
+#![allow(unused)]
+
+use std::time::{Duration, Instant};
+
+// Recommended values from https://datatracker.ietf.org/doc/html/draft-ietf-rmcat-gcc-02#section-5
+const OBSERVED_BIT_RATE_SMOOTHING_FACTOR: f64 = 0.95;
+const BETA: f64 = 0.85;
+
+/// A rate controller that estimates a suitable send bitrate.
+///
+/// Inputs to the rate controller are:
+/// * The observed received bitrate(via TWCC feddback).
+/// * RTT.
+/// * Congestion estimates from the delay controller.
+struct RateController {
+    state: State,
+
+    estimated_bitrate: f64,
+    min_bitrate: f64,
+    max_bitrate: f64,
+
+    /// The last observed bitrate calculated based on TWCC data.
+    last_observed_bitrate: Option<f64>,
+    /// The averaged observed bitrate when we have been in the decrease state.
+    averaged_observed_bitrate: MovingAverage,
+    /// The last time we updated the estimated bitrate.
+    last_estimate_update: Instant,
+    // Last RTT estimate in micro-seconds
+    last_rtt_us: Option<f64>,
+}
+
+impl RateController {
+    pub(super) fn new(
+        estimated_bitrate: u64,
+        min_bitrate: u64,
+        max_bitrate: u64,
+        now: Instant,
```

Comment:
As mentioned in the other PR, I think it would be valuable to create DataRate and DataSize structs to wrap bitrates and sizes to avoid mistakes with bitrate units.